### PR TITLE
Directory adjustments

### PR DESCRIPTION
**Pull request by amotl**
As suggested by @mechanomi, we wanted to rename two directories before cutting the next release.

- `src` -> `common-build`: As this is what we talk about in conversation so it's less ambiguous.
- `docs` -> `demo-docs`: This Sphinx project only exists as a demo project that can be used to test the other tools within this very repository.

### CODE REVIEW

File: demo-docs/Makefile [hunk 1]
```diff
@@ -23,13 +23,13 @@
 
 .EXPORT_ALL_VARIABLES:
 
-DOCS_DIR      := docs
+DOCS_DIR      := demo-docs
```
Comment (amotl) [inline demo-docs/Makefile:26]: That might be the only thing which might create a woe when used within downstream repositories?

Comment (amotl) [reply demo-docs/Makefile:26]: We might want to add a bit of heuristics to discover the "docs" directory from a list of `[docs, doc, demo-docs]`?

Comment (mechanomi) [reply demo-docs/Makefile:26]: oh good point, yeah

**Comment by mechanomi** [comment]
@amotl lemme know when this is ready for review again. atm I am presuming that you intend to add docs directory discovery to the Makefile so am holding off

**Review by infoverload — APPROVED** [review]

**Comment by amotl** [comment]
Hi Naomi,

please let me know if you think we should update and bring in this change before making #64 hit `main`.

With kind regards,
Andreas.

**Comment by norosa** [comment]
@amotl I think this is ready to go. just need to resolve the conflicts

**Comment by amotl** [comment]
I will open a new PR for this instead of going through the hassle of rebasing with directory renames.

File: demo-docs/Makefile (hunk 1 above)
Comment (amotl) [reply demo-docs/Makefile:26]: Maybe I was overthinking this back then. `demo-docs/Makefile` will actually be just a blueprint, so there will be no need to provide a directory autodiscovery here anyway, right?

**Comment by amotl** [comment]
This patch got superseded by #67.

**Pull request closed without merging**